staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/envelope.go

```diff
@@ -61,12 +58,16 @@ type envelopeTransformer struct {
 // the data items they encrypt. A cache (of size cacheSize) is maintained to store the most recently
 // used decrypted DEKs in memory.
 func NewEnvelopeTransformer(envelopeService Service, cacheSize int, baseTransformerFunc func(cipher.Block) value.Transformer) (value.Transformer, error) {
-	if cacheSize == 0 {
-		cacheSize = defaultCacheSize
-	}
-	cache, err := lru.New(cacheSize)
-	if err != nil {
-		return nil, err
+	var (
+		cache *lru.Cache
+		err   error
+	)
+
+	if cacheSize > 0 {
```

Comment:
This is a behavior change - previously `cacheSize == 0` meant default cacheSize.
It wasn't possible to run this transformer without a cache.

Now, cache is optional, which complicates the code.

Does it ever make sense to run without a cache?

Reply:
1. Security conscious users may choose to trade performance for security by opting out of cache. This is consistent with the best practices of not keeping secrets in memory after use.
2. When users explicitly set CacheSize to 0 (prior to this PR) the resulting CacheSize will be 1000 - probably not what users would expect.

Reply:
> Security conscious users may choose to trade performance for security by opting out of cache.

Doesn't that mean a list API call makes potentially thousands (or even tens or hundreds of thousands) of KMS calls? Is that even workable?

Reply:
Added validation check for the case where cachesize is set to 0. Such condition will result in a validation error.